crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -968,7 +968,21 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                         .wait_blocking(&version_id)
                         .ok_or(ResolveError::Unregistered)?;
 
-                    return Ok(Dependencies::Available(Vec::default()));
+                    // If a package has a marker, add a dependency from it to the
+                    // same package without markers.
```

Comment:
Similar to the comment below, there may be a better way to solve this. I have a feeling we might want a separate PubGrubPackageInner marker type.

Reply:
I have that same feeling, especially after you added the `Extra` variant.